homeassistant/components/tuya/base.py

```diff
@@ -235,6 +239,13 @@ def find_dpcode(
                         )
                     ):
                         continue
+
+                    # Workaround for BHT-002 Thermostat devices which use the base 2
+                    # instead of 10 during scaling of the integer value (but not the
+                    # step). This is a violation from the specification.
+                    if self.device.product_id == "IAYz2WK1th0cMLmL":
```

Comment:
This is device specifics in our code, which we don't want. We process the data correctly, this is a know issue with this device, please contact the device manufacturer or Titus, as the data they provide is incorrect.

../Frenck

Reply:
Well. The the device is produced by a Chinese OEM manufacturer. They are not possible to get in touch with, and will probably never update the firmware only because the thermostat won't work in home assistant. Smart Home (the offical app) already do have a workaround similar to what is done in this pull request. Thus my (and thousands of other home assistant users) only options are:

* Throw our thermostats and by new ones (in my case four of them)
* Or, stop use the offical Tuya integration and use a custom one
* Or, hack the files in the offical Tuya integration (which seems to be the most common approach)

For me it feels like a no brainer to add these two workaround lines in the code and make all users happy. It wont break anything. 

I have been working as a software developer and architect in the consumer, medtech, defense and automation industry for 20 years and making workarounds for devices not following specifications is extremely common.

Reply:
> For me it feels like a no brainer to add these two workaround lines in the code and make all users happy. It wont break anything.

We don't want device specific code/exceptions in Home Assistant. It will end up with hundreds of exceptions making code unmaintainable in the long run.

These things should ideally be fixed in the device or in the Tuya cloud (as they now send incorrect values). Alternatively, as a last resort, quirk handling could be done in the library that handles communication. The latter is also maintained by Tuya themselves publicly on GitHub.

../Frenck

Reply:
Well, Tuya cloud is out of my control. Quick fix in the Tuya communication library on GitHub is according to me even uglier than doing the fix in home assistant. 

I could make the workaround a little less "hacky" in the code if you are interested. We could for example have a specific python file with a dictionary of device specific settings that overrides the default settings. This would make it more maintainable.

Reply:
As said now three times, we don't want this type of device specific code in Home Assistant. I have already supplied the acceptable alternatives.

../Frenck